File: packages/java/typescript-generator/src/test/java/com/vaadin/hilla/parser/plugins/nonnull/nonnullapi/NonNullApiEndpoint.java
/*
 * Copyright 2000-2025 Vaadin Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package com.vaadin.hilla.parser.plugins.nonnull.nonnullapi;

import java.util.Map;
import java.util.Optional;
import java.util.Set;

import com.vaadin.hilla.parser.plugins.nonnull.nonnullapi.subpackage.SubPackageDependency;
import com.vaadin.hilla.parser.testutils.annotations.Endpoint;

@Endpoint
public class NonNullApiEndpoint {
    public Dependency defaultMethod(String param) {
        return null;
    }

    public Map<String, Set<Dependency>> nestedSignatureMethod(
            Set<Dependency> param) {
        return null;
    }

    @NullableMethod
    public Dependency nullableMethod(@NullableParameter String param) {
        return null;
    }

    public Map<String, @NullableSignature Set<@NullableSignature Dependency>> nullableNestedSignatureMethod(
            Set<@NullableSignature Dependency> param) {
        return null;
    }

    @NullableSignature
    public Dependency nullableSignature(@NullableSignature String param) {
        return null;
    }

    public static class Dependency {
        public String defaultField;
        @NullableField
        public String nullableField;
        @NullableSignature
        public String nullableSignatureField;
    }

    public Optional<String> optionalMethod(Optional<String> opt) {
        return opt;
    }

    public SubPackageDependency subPackageMethod(SubPackageDependency entity) {
        return entity;
    }

    public static class Generic<T> {
        public T genericField;
    }

    public Generic<String> genericMethod(Generic<String> generic) {
        return generic;
    }

    public Generic<@NullableSignature String> genericNullableMethod(
            Generic<@NullableSignature String> generic) {
        return generic;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/nonnullapi/snapshots/NonNullApiEndpoint.ts
import type { EndpointRequestInit } from '@vaadin/hilla-frontend';
import type Dependency from './com/vaadin/hilla/parser/plugins/nonnull/nonnullapi/NonNullApiEndpoint/Dependency.js';
import type Generic from './com/vaadin/hilla/parser/plugins/nonnull/nonnullapi/NonNullApiEndpoint/Generic.js';
import type SubPackageDependency from './com/vaadin/hilla/parser/plugins/nonnull/nonnullapi/subpackage/SubPackageDependency.js';
import client from './connect-client.default.js';

export async function defaultMethod(param: string, init?: EndpointRequestInit): Promise<Dependency> {
  return client.call('NonNullApiEndpoint', 'defaultMethod', { param }, init);
}

export async function genericMethod(generic: Generic<string>, init?: EndpointRequestInit): Promise<Generic<string>> {
  return client.call('NonNullApiEndpoint', 'genericMethod', { generic }, init);
}

export async function genericNullableMethod(
  generic: Generic<string | undefined>,
  init?: EndpointRequestInit,
): Promise<Generic<string | undefined>> {
  return client.call('NonNullApiEndpoint', 'genericNullableMethod', { generic }, init);
}

export async function nestedSignatureMethod(
  param: Array<Dependency>,
  init?: EndpointRequestInit,
): Promise<Record<string, Array<Dependency>>> {
  return client.call('NonNullApiEndpoint', 'nestedSignatureMethod', { param }, init);
}

export async function nullableMethod(
  param: string | undefined,
  init?: EndpointRequestInit,
): Promise<Dependency | undefined> {
  return client.call('NonNullApiEndpoint', 'nullableMethod', { param }, init);
}

export async function nullableNestedSignatureMethod(
  param: Array<Dependency | undefined>,
  init?: EndpointRequestInit,
): Promise<Record<string, Array<Dependency | undefined> | undefined>> {
  return client.call('NonNullApiEndpoint', 'nullableNestedSignatureMethod', { param }, init);
}

export async function nullableSignature(
  param: string | undefined,
  init?: EndpointRequestInit,
): Promise<Dependency | undefined> {
  return client.call('NonNullApiEndpoint', 'nullableSignature', { param }, init);
}

export async function optionalMethod(opt: string | undefined, init?: EndpointRequestInit): Promise<string | undefined> {
  return client.call('NonNullApiEndpoint', 'optionalMethod', { opt }, init);
}

export async function subPackageMethod(
  entity: SubPackageDependency,
  init?: EndpointRequestInit,
): Promise<SubPackageDependency> {
  return client.call('NonNullApiEndpoint', 'subPackageMethod', { entity }, init);
}
